lib/src/main/java/com/fairportrobotics/frc/poe/CameraTracking/AprilTags.java

```diff
@@ -0,0 +1,83 @@
+
+    public AprilTags(String...cameraNames) {
+        for(String cameraName : cameraNames) {
+            cameras.add(new PhotonCamera(cameraName));
+        }
+    }
+
+    // Method to get a list of all targets from all cameras
+    public List<PhotonTrackedTarget> getTargets() {
+        List<PhotonTrackedTarget> targets = new ArrayList<>();
+
+        for(PhotonCamera camera : cameras) {
+            for(PhotonTrackedTarget target : camera.getLatestResult().getTargets()) {
+                boolean toAdd = true;
+                // check if the target is already in the list and if it has a higher pose ambiguity
+                for(PhotonTrackedTarget arTarget : targets) {
+                    if(target.getFiducialId() == arTarget.getFiducialId() && target.getPoseAmbiguity() < arTarget.getPoseAmbiguity()) {
+                        // if it does, remove it from the list
+                        targets.remove(arTarget);
+                        break;
+                    } else if(target.getFiducialId() == arTarget.getFiducialId()) {
+                        // if the target is already in the list and has the same ID, don't add it
+                        toAdd = false;
+                        break;
+                    }
+                }
+
+                if(toAdd) targets.add(target);
+            }
+            targets.addAll(camera.getLatestResult().getTargets());
+        }
+
+        return targets;
+    }
+
+    // method to check if any camera has any targets
+    public boolean hasTargets() {
+        for(PhotonCamera camera : cameras) {
+            // if a camera has targets, return true
+            if(camera.getLatestResult().hasTargets()) return true;
+        } 
+        
+        return false;
+    }      
+
+    // method to get the closest target
+    public PhotonTrackedTarget getClosestTarget() {
+        if(!hasTargets()) return null;
+        PhotonTrackedTarget closestTarget = null;
+        for(PhotonTrackedTarget target : getTargets()) {
+            if(closestTarget == null || calculateDistance(target.getBestCameraToTarget()) > calculateDistance(target.getBestCameraToTarget()) ) closestTarget = target;
+        }
+        return closestTarget;
+    }
+
+    // method to get the target with the lowest pose ambiguity
+    public PhotonTrackedTarget getLowestAmbiguityTarget() {
+        if(!hasTargets()) return null;
+        PhotonTrackedTarget lowestAmbiguity = null;
+        for(PhotonTrackedTarget target : getTargets()) {
+            // check if the target has a lower pose ambiguity than the current lowest ambiguity target
+            if(lowestAmbiguity == null || lowestAmbiguity.getPoseAmbiguity() > target.getPoseAmbiguity()) lowestAmbiguity = target;
+        }
+        return lowestAmbiguity;
+    }
+
+    private double calculateDistance(Transform3d t) {
+        return Math.sqrt(t.getX()*t.getX() + t.getY()*t.getY() + t.getZ()*t.getZ());
```

Comment:
Remove Z dimension

And

This only calculates the distance from the passed in object and itself.

Camera is at 0,0 so try this

```suggestion
        return Math.sqrt(Math.pow(t.getX()-0,2) + Math.pow(t.getY()-0, 2));
```

Or

```suggestion
        return Math.sqrt(t.getX()*t.getX() + t.getY()*t.getY());
```